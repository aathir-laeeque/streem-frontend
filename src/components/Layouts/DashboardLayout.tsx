import { openOverlayAction } from '#components/OverlayContainer/actions';
import { OverlayNames } from '#components/OverlayContainer/types';
import { useTypedSelector } from '#store';
import { setInternetConnectivity } from '#store/extras/action';
import { DEFAULT_SESSION_TIMEOUT_IN_MIN } from '#utils/constants';
import { setIdle } from '#views/Auth/actions';
import { ssoLogout } from '#views/Auth/saga';
import { UserType } from '#views/UserAccess/ManageUser/types';
import React, { FC, useEffect } from 'react';
import { useIdleTimer } from 'react-idle-timer';
import { useDispatch } from 'react-redux';
import styled from 'styled-components';
import { useMsal } from '@azure/msal-react';
import { useLocation } from '@reach/router';

const Layout = styled.div.attrs<React.HTMLAttributes<HTMLDivElement>>({
  className: 'main-layout-view',
})`
  display: grid;
  grid-template-areas:
    'header header'
    'nav-menu workarea';
  grid-template-rows: auto 1fr;
  grid-template-columns: auto 1fr;
  flex: 1;
  background-color: #fff;
`;

const DashboardLayout: FC<React.HTMLAttributes<HTMLDivElement>> = ({ children, ...rest }) => {
  const dispatch = useDispatch();
  const location = useLocation();
  const { instance } = useMsal();
  const { isIdle, isLoggedIn, settings, userType } = useTypedSelector((state) => state.auth);

  useEffect(() => {
    if (isIdle && isLoggedIn) {
      dispatch(
        openOverlayAction({
          type: OverlayNames.SESSION_EXPIRE,
        }),
      );
    }

    const handleConnectivityChange = (connected: boolean) => {
      dispatch(setInternetConnectivity({ connected }));
    };

    window.addEventListener('online', () => handleConnectivityChange(true));
    window.addEventListener('offline', () => handleConnectivityChange(false));

    return () => {
      window.removeEventListener('online', () => handleConnectivityChange(true));
      window.removeEventListener('offline', () => handleConnectivityChange(false));
    };
  }, []);

  const handleOnIdle = () => {
    const { pathname } = location;
    const splitPath = pathname.split('/');
    if (splitPath?.[1] === 'reports') {
      reset();
    } else if (!isIdle && isLoggedIn) {
      if (userType === UserType.AZURE_AD) {
        ssoLogout(instance);
      }
      dispatch(setIdle(true));
      dispatch(
        openOverlayAction({
          type: OverlayNames.SESSION_EXPIRE,
        }),
      );
    }
  };

  const { reset } = useIdleTimer({
    timeout: 1000 * 60 * (settings?.sessionIdleTimeoutInMinutes || DEFAULT_SESSION_TIMEOUT_IN_MIN),
    onIdle: handleOnIdle,
    debounce: 500,
  });

  return <Layout {...rest}>{children}</Layout>;
};

export default DashboardLayout;
