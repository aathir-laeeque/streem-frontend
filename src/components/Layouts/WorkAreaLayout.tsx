import React, { FC } from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  grid-area: workarea;

  > div {
    height: 100%;
  }

  background-color: #ecedf1;
  border-top-left-radius: 20px;
  border-top: 1.5px solid rgb(184, 184, 184);
  height: inherit;
  padding: 10px 8px 0 12px;
`;

const ContentArea = styled.div`
  background-color: #ffffff;
  border-radius: 5px;
  box-shadow: 0 5px 8px 0 rgba(0, 0, 0, 0.1);
  height: inherit;
`;

const WorkAreaLayout: FC = ({ children }) => {
  return (
    <Wrapper>
      <ContentArea>{children}</ContentArea>
    </Wrapper>
  );
};

export default WorkAreaLayout;
