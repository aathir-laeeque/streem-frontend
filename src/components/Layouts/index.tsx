export { default as DashboardLayout } from './DashboardLayout';
export { default as WorkAreaLayout } from './WorkAreaLayout';
