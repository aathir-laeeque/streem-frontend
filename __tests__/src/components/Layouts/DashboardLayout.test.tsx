import { DashboardLayout } from '#components/Layouts';
import React from 'react';
import { create, ReactTestRenderer } from 'react-test-renderer';

describe('<DashboardLayout />', () => {
  let snapshot: ReactTestRenderer;

  beforeAll(() => {
    const component = <DashboardLayout />;
    snapshot = create(component);
  });

  it('should render and match the snapshot', () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
