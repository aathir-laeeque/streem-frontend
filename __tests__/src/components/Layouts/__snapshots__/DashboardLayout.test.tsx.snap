// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardLayout /> should render and match the snapshot 1`] = `
<div
  className="sc-AxjAm dRhhfm"
/>
`;
